### ISSUE TITLE
Multi-species operation

### ISSUE THREAD

**Issue by luoxixi666** [lorrainea/CNEFinder#18]
Hello, I would like to ask CNEfinder if it can find non-coding sequences that are conserved in many species at one time, I have about a dozen species, how to operate it, thank you very much

**Comment by lorrainea** [comment]
Hi, currently CNEFinder only works for 2 sequences at a time. You could do an all by all pairwise comparison then output the actual CNEs using the positions and lengths in the output file. You can then identify shared CNEs amongst all of the sequences by finding those CNEs that exist in all sequences with some given similarity threshold.

**Comment by luoxixi666** [comment]
> Hi, currently CNEFinder only works for 2 sequences at a time. You could do an all by all pairwise comparison then output the actual CNEs using the positions and lengths in the output file. You can then identify shared CNEs amongst all of the sequences by finding those CNEs that exist in all sequences with some given similarity threshold.

Thank you very much for your answer, but if you want to compare the whole genome of two species, what should CNEfinder do, and do you want to merge all the chromosomes of the whole genome into one large chromosome? But I always got an error when I tried last time, I don't know if it's a problem with my method, can you give me some guidance, thank you so much! ~

**Comment by lorrainea** [comment]
Hi, yes it is possible to compare the whole genome of two sequences. For each sequence just concatenate all the chromosomes into one sequence. Use the ID names you give for these sequences as input parameters for -y and -z, then parameters -a, -b, -c and -d should be set as 0 to compare the entire sequences.

This page may also help explain the parameters more: https://github.com/lorrainea/CNEFinder/wiki/Running-CNEFinder

If you still get an error please let me know what command you are attempting to run.

**Comment by luoxixi666** [comment]
> Hi, yes it is possible to compare the whole genome of two sequences. For each sequence just concatenate all the chromosomes into one sequence. Use the ID names you give for these sequences as input parameters for -y and -z, then parameters -a, -b, -c and -d should be set as 0 to compare the entire sequences.
> 
> This page may also help explain the parameters more: https://github.com/lorrainea/CNEFinder/wiki/Running-CNEFinder
> 
> If you still get an error please let me know what command you are attempting to run.

I'm sorry I want to ask what the specific running command is, do you still need exome coordinates, can you give me an example, thank you

**Comment by lorrainea** [comment]
Hi, if you look at the examples in the link above that may help. The command you are looking to use is something similar to:

./cnef  -r <ref_sequence_name>  -q <query_sequence_name>  -e <ref_exons_filename>  -f <query_exons_filename> -y <ref_ID_name> -z <query_ID_name>  -a 0 -b 0 -c 0 -d 0 -t <threshold_required> -l <minimum_length_required> -o <output_filename>

Those parameters between <> need to be defined by you depending on your filenames and required CNE properties. If you do not have any exon coordinates then you can create an empty file and use that as input.

**Comment by luoxixi666** [comment]
> Hi, if you look at the examples in the link above that may help. The command you are looking to use is something similar to:
> 
> ./cnef -r <ref_sequence_name> -q <query_sequence_name> -e <ref_exons_filename> -f <query_exons_filename> -y <ref_ID_name> -z <query_ID_name> -a 0 -b 0 -c 0 -d 0 -t <threshold_required> -l <minimum_length_required> -o <output_filename>
> 
> Those parameters between <> need to be defined by you depending on your filenames and required CNE properties. If you do not have any exon coordinates then you can create an empty file and use that as input.

Does this ref_ID_name refer to the first line of names in the genome-wide file? For example> Chr1 and that? Because I've now combined all the chromosomes of the aligned species into one big chromosome file, which is equivalent to only one 'chromosome', but I just tried to do it and it seems to get an error, can you give me some more detailed advice and guidance, thank you very much! Here's the command I run: cnef -r merged1_genome.fa -q merged2_genome.fa -e merged1_exons.tsv -f merged2_exons.tsv -y Chr1 -z Chr1 -a 0 -b 0 -c 0 -d 0 -t 0.95 -l 150 -o whole.output

**Comment by lorrainea** [comment]
Sure let's take this step by step. If the following command is what you use:

./cnef -r merged1_genome.fa -q merged2_genome.fa -e merged1_exons.tsv -f merged2_exons.tsv -y Chr1 -z Chr1 -a 0 -b 0 -c 0 -d 0 -t 0.95 -l 150 -o whole.output

Then this means:

merged1_genome.fa consists of a fasta file with one sequence and one ID with the name Chr1
merged2_genome.fa consists of a fasta file with one sequence and one ID with the name Chr1
merged1_exons.tsv and merged2_exons.tsv are empty files if you have no exons.

Let me know what error you get.

**Comment by luoxixi666** [comment]
> Sure let's take this step by step. If the following command is what you use:
> 
> ./cnef -r merged1_genome.fa -q merged2_genome.fa -e merged1_exons.tsv -f merged2_exons.tsv -y Chr1 -z Chr1 -a 0 -b 0 -c 0 -d 0 -t 0.95 -l 150 -o whole.output
> 
> Then this means:
> 
> merged1_genome.fa consists of a fasta file with one sequence and one ID with the name Chr1 merged2_genome.fa consists of a fasta file with one sequence and one ID with the name Chr1 merged1_exons.tsv and merged2_exons.tsv are empty files if you have no exons.
> 
> Let me know what error you get.

Thank you for your answer, I have now successfully run the whole genome CNEfinder, I found that when he runs, it will generate a tmp file, which is cut into 25 parts, and it runs very slowly, does this software have thread settings similar to CPU, is there any way to speed up the operation, thank you

**Comment by lorrainea** [comment]
You can use parameter -T for multiple threads.

**Comment by luoxixi666** [comment]
> You can use parameter -T for multiple threads.

Hello, I reported an error when I was running, the error is: Segmentation fault (core dumped) seems to say that the memory is insufficient, but I checked that the server memory is still more than 500G, is the file obtained by the whole genome run very large, how to deal with this situation?

**Comment by lorrainea** [comment]
The CNEs are stored in internal memory and these depend on the sequences input. Are you using a number of threads that exist on your system?

**Comment by luoxixi666** [comment]
> The CNEs are stored in internal memory and these depend on the sequences input. Are you using a number of threads that exist on your system?

Sorry I didn't understand what you meant, here's the command line I run, please see if there are any problems: cnef -r 1_merged_genome.fa -q 2.merged_genome.fa -e 1_exons.tsv -f 2_exons.tsv -y Chr1 -z Chr1 -a 0 -b 0 -c 0 -d 0 -t 0.95 -l 150 -T 20 -o all.output

**Comment by lorrainea** [comment]
Could you please also send your input files so I can look into this in detail.

**Comment by luoxixi666** [comment]
> Could you please also send your input files so I can look into this in detail.

Of course! Due to the limitation of the upload file size, the folder is part of the content of my input file, the genome file is about 2G, it is to merge all the chromosomes into a "Chr1", in order to run the whole genome, can you help me see and analyze the cause of the error, thank you very much!
[Uploading examples.zip…]()

**Comment by lorrainea** [comment]
Please use an external source to upload the files if they cannot be uploaded on Github and provide the link here. Thank you.
